### DISCUSSION TITLE
What commerce-adjacent patterns should be bundled with Woo Core

### DISCUSSION THREAD

**Discussion by nerrad** [woocommerce/woocommerce#44989] [WooCommerce, Blocks, and Site Editing]
One of the things we're working toward as a part of the Store Editing work is ensuring WooCommerce works great out of the box with _any_ block theme (not just block themes that explicitly support WooCommerce). There will still always be the need for themes that explicitly support commerce, but by providing support for a storefront to _every_ block theme this increases the value of WooCommerce across the WordPress ecosystem as well as making it easier for theme developers and designers to enhance or build-off of what Woo provides.

To accomplish this, one of the areas we intend to start on soon is bundling a really great set of what I'm calling "commerce-adjacent" patterns in WooCommerce. 

> tldr;
> I'm looking for feedback from those who build themes or customize WooCommerce sites about your thoughts about Woo including these types of patterns. What you would be happy to see, what are you hoping we'd stay away from? And why?

## What are commerce-adjacent patterns?

These are patterns that are used in a commerce context but aren't dependent on using specific WooCommerce blocks or being connected to WooCommerce data. They will primarily be constructed from WP core blocks. Typically, because the patterns are not directly linked to products, they would need to be manually updated once inserted (adding links, product names, images etc) and also updated when relevant data changes.

Some examples:
_Note these are just to **illustrate** what I mean by commerce-adjacent. They are not proposals of exactly what would be bundled in WooCommerce core. Also, thanks to @anarieldesign for these examples which come from the [Basti theme](https://demo.anarieldesign.com/basti/ecommerce-patterns/)._

Product cards showcasing specific products.
![Screenshot 2022-08-24 at 13 07 48](https://user-images.githubusercontent.com/1429108/186450684-f984055a-828d-4a02-948e-c5e58f896a84.png)

Customized reviews
![Screenshot 2022-08-24 at 13 08 00](https://user-images.githubusercontent.com/1429108/186451056-39506c19-5821-4ff0-9c6a-95b0e6262202.png)

Sale advertisement/callout.
![Screenshot 2022-08-24 at 13 08 08](https://user-images.githubusercontent.com/1429108/186451501-c73c1fca-02aa-4ef7-aac9-6ea46952aec3.png)

## Looking for your feedback!

We want to make sure we're:

- bundling the most _useful_ patterns.
- leaving room for creativity from the ecosystem (i.e. we're not going to bundle for _every_ use case.
- exploring ideas on how we could potentially make it easier for block theme developers to build off of these patterns or incorporate them in their designs.

So with that in mind, here's a few questions to kick-start the discussion:

- What commerce-adjacent type patterns would you be really excited to see bundled with WooCommerce core?
- What are you hoping we'll _not_ do as a part of this initiative?
- Why?

**Comment by ellenbauer** [comment]
This is very interesting and exciting to read. I think it would be great to focus on patterns that are most widely used or somehow serve a general purpose for eCommerce websites. 

Making it easier for block themes to build on top of these sounds great. This way we can build faster and don't need to reinvent the wheel.

Of the top of my head I can think of store testimonials, product sliders, mega menus, featured categories, banners, newsletter subscribe forms even, single product image galleries. Plus I don't know how the Woo blocks may already solve this but a general way of having the account work would be great. As with checkout and cart it would be best to have them always work the same way and out of the box.

I hope this is helpful somehow.
